compiler/cmdlinehelper.nim

```diff
@@ -0,0 +1,83 @@
+## Helpers for binaries that use compiler passes, eg: nim, nimsuggest, nimfix
+
+# TODO: nimfix should use this; currently out of sync
+
+import
+  compiler/[options, idents, nimconf, scriptconfig, extccomp, commands, msgs, lineinfos, modulegraphs, condsyms],
+  std/os
+
+type ProgBase* = ref object of RootObj
+  name*: string
+  suggestMode*: bool
+
+method processCmdLine(self: ProgBase, pass: TCmdLinePass, cmd: string; config: ConfigRef) {.base.} =
+  doAssert false
+
+method mainCommand(self: ProgBase, graph: ModuleGraph) {.base.} =
+  doAssert false
+
+proc initDefinesProg*(self: ProgBase, conf: ConfigRef) =
+  condsyms.initDefines(conf.symbols)
+  defineSymbol conf.symbols, self.name
+
+proc processCmdLineAndProjectPath*(self: ProgBase, conf: ConfigRef) =
```

Comment:
Instead of this OOP approach, give ``processCmdLineAndProjectPath`` two callbacks or - probably better- make it a template.

Reply:
done with 2 callbacks